llvm/test/CodeGen/AMDGPU/attr-amdgpu-max-num-workgroups.ll

```diff
@@ -54,16 +81,25 @@ attributes #4 = {"amdgpu-max-num-workgroups"="1024,1024,1024"}
 
 ; CHECK: - .args:
 ; CHECK:        .max_flat_workgroup_size: 1024
+; CHECK-NEXT:   .max_num_workgroups_x: 0
```

Comment:
What does 0 here mean?

Reply:
It doesn't say that. It states "Clang only emits this attribute when all the three numbers are >= 1." Which doesn't really make sense.

0 doesn't make sense as a value

Reply:
It does say that: "The max number of launched work-groups in the X, Y, and Z dimensions. Each number must be >=1."

Reply:
In trivial questions, is the hardware maximum for workgroups actually `uint32_t_max` in all dimensions, or is it `uint32_t_max / [wave size]`?

Reply:
That's on the emitted directive, it's not directly documented on the IR attribute.

I think I've seen that it's lower in the y and z dimensions but I don't remember where and can't find documentation for it